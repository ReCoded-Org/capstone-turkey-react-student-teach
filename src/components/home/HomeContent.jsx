import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Question from './Question';
// eslint-disable-next-line import/no-named-as-default
import { fetchQuestions } from '../../redux/features/questionsSlice';
import AddQuestion from '../modals/AddQuestion/AddQuestion';
import Hands from '../../assets/hands/Hands';
import Avatar from '../../assets/images/avatar.jpg';
import CheckAuth from '../modals/checkAuth/CheckAuth';

function HomeContent() {
  const [addQuestionModal, setAddQuestionModal] = useState(false);
  const signIn = useSelector((state) => state.signIn);
  const isSuccess = signIn.user.status;
  const isUser = signIn.user.userInfo;
  const isUserSignedUp = signIn.signUp.isSignedUp;
  const allQuestions = useSelector((state) => state.questions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchQuestions());
  }, [dispatch]);
  const darkMode = useSelector((state) => state.darkModeReducer.darkMode);
  return (
    <div
      className={`${
        darkMode ? 'bg-secondaryDark' : 'bg-zinc-100 '
      } pt-5 pb-[5rem]`}
    >
      <div
        className={` container mx-auto w-full pt-8 rounded-md transition-all ease-in-out duration-300  text-center lg:text-left ${
          darkMode ? 'bg-primaryDark' : 'bg-white'
        } ${darkMode ? 'text-white' : 'text-black'} `}
      >
        <section className="flex min-h-[30vh] lg:min-h-[50vh]">
          <div className="min-w-[30vw]">
            <h2 className="text-5xl sm:text-7xl text-orange mb-5 lg:mb-0">
              Student Teach
            </h2>
            <h4 className="font-semibold text-base sm:text-2xl mt-2 sm:mt-9 py-0.5">
              We grow up by helping each other.
            </h4>
            <p className="mt-2 sm:mt-9 text-sm sm:text-xl  py-0.5 w-12/12 lg:w-10/12 ">
              This is a platform where you can ask your questions and help
              people find answers.
            </p>
            <p className="mt-2 sm:mt-9 text-sm sm:text-xl  py-0.5 w-12/12 lg:w-10/12 ">
              There are hundreds of categories in which you can ask your
              questions and you can learn by traveling through them.
            </p>
          </div>

          <div className=" hidden min-w-[50vw] lg:flex flex-col items-center">
            <Hands
              css={`h-[20rem] w-fit  ${darkMode ? 'fill-white' : 'fill-black'}`}
            />
            <button
              type="button"
              className="bg-orange rounded-md text-xs w-44 h-10 scale-125 text-[#FFFFFF] inline-block bottom-[20px] xl:left-[40%] lg:left-[20%] hover:scale-[1.35] ease-in-out transition-all"
              onClick={() => setAddQuestionModal(true)}
            >
              Ask Question +
            </button>
          </div>
        </section>
        <button
          type="button"
          className=" lg:hidden mt-10 text-xs ml-5 sm:text-3xl mx-auto bg-orange rounded-md sm:w-[14rem] sm:h-[4rem] w-[8rem] h-[2rem] text-[#FFFFFF] absolute inline-block  left-[30%]"
          onClick={() => setAddQuestionModal(true)}
        >
          Ask Question +
        </button>
        <div className="mt-32 md:mr-7 lg:mr-10 pb-10">
          {allQuestions.questions
            ?.slice(
              allQuestions.questions.length - 10,
              allQuestions.questions.length,
            )
            .reverse()
            .map((q) => (
              <Question
                // eslint-disable-next-line no-underscore-dangle
                key={q._id}
                // eslint-disable-next-line no-underscore-dangle
                id={q._id}
                question={q.title}
                student={q.student}
                profileImage={Avatar}
                answer={
                  q.comments[0]
                    ? q.comments[0]
                    : 'Give an answer to the question'
                }
              />
            ))}
        </div>
        {isSuccess === 'success' ||
        signIn.signUp.status === 'success' ||
        isUser.firstName ||
        isUserSignedUp.firstName ? (
          <AddQuestion open={addQuestionModal} setOpen={setAddQuestionModal} />
        ) : (
          <CheckAuth
            label="Sign in to add question."
            open={addQuestionModal}
            setOpen={setAddQuestionModal}
          />
        )}
      </div>
    </div>
  );
}

export default HomeContent;
