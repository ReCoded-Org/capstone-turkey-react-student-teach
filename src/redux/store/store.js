import { configureStore } from '@reduxjs/toolkit';
import addQuestionSlice from '../features/addQuestionSlice';

const store = configureStore({
  reducer: {
    addQuestionReducer: addQuestionSlice,
    signIn: userSlice,
  },
});

export default store;
