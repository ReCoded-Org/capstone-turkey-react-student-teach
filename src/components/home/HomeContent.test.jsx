import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import HomeContent from './HomeContent';
import store from '../../redux/store/store';

it('Render when text of no HomeContent found removes', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <HomeContent />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
