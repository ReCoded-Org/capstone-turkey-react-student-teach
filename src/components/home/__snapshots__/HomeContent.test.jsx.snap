// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render when text of no HomeContent found removes 1`] = `
<div
  className="bg-zinc-100  py-5"
>
  <div
    className=" container mx-auto w-full pt-8 rounded-md transition-all ease-in-out duration-300  text-center lg:text-left bg-white text-black "
  >
    <section
      className="flex min-h-[30vh] lg:min-h-[50vh]"
    >
      <div
        className="min-w-[30vw]"
      >
        <h2
          className="text-5xl sm:text-7xl text-orange mb-5 lg:mb-0"
        >
          Student Teach
        </h2>
        <h4
          className="font-semibold text-base sm:text-2xl mt-2 sm:mt-9 py-0.5"
        >
          We grow up by helping each other.
        </h4>
        <p
          className="mt-2 sm:mt-9 text-sm sm:text-xl  py-0.5 w-12/12 lg:w-10/12 "
        >
          This is a platform where you can ask your questions and help people find answers.
        </p>
        <p
          className="mt-2 sm:mt-9 text-sm sm:text-xl  py-0.5 w-12/12 lg:w-10/12 "
        >
          There are hundreds of categories in which you can ask your questions and you can learn by traveling through them.
        </p>
      </div>
      <div
        className=" hidden min-w-[50vw] lg:flex flex-col items-center"
      >
        <div>
          <svg
            className="h-[20rem] w-fit  fill-black"
            height="474.66667"
            version="1.0"
            viewBox="0 0 702 356"
            width="936"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M156 54c-4.7 1-14.8 4.2-22.5 7-23 8.5-30.5 11-40.5 13.5-9.8 2.5-36.9 7.5-52.2 9.6-4.9.7-8.8 1.6-8.8 2.1 0 1.1-1.2 1.2 20.6-2.2 26.8-4.1 47.4-9 65.4-15.6 33.2-12.1 39.5-13.8 50.5-13.8 11.5-.1 18 1.7 36.4 9.6 37.3 16.2 64.9 33.4 73.1 45.5 3.1 4.6 3.2 5.1 2.6 12.3-.3 4.1-1.7 12.8-3.1 19.2-1.4 6.5-2.5 12.9-2.5 14.2 0 2.1-.7 2.7-4 3.6-2.1.6-4.6 1.7-5.4 2.5-1 1-2.5 1.3-4.8.8-5.7-1.1-8.1-2.6-13.3-8.2l-5-5.3v-8.7c0-5.6.6-10.4 1.8-14 1-2.9 1.6-5.6 1.3-5.9-.3-.3-2.8 1.5-5.4 4-13.2 12.3-19.1 9.9-19.3-7.7 0-9-1-9.5-2.4-1.4-.3 2.1-1.5 4.9-2.7 6.4-1.9 2.4-2.7 2.7-6.4 2.2-4.6-.5-15.5-7.3-14.8-9.2.7-1.6-2.3-4-9.1-7.5-11-5.5-12.6-3.3-3.7 5.3 11.8 11.5 26 21.4 44.7 31.1 10.9 5.6 14 7.8 18.8 12.9 9.5 10.4 18.7 22.7 19.4 25.9.4 2.2.3 2.9-.5 2.4-.7-.4-2.1 1.2-3.6 4.2-1.4 2.7-3.7 6.3-5.1 8-3.1 3.7-2.7 4.9 1.9 5.7 5.8 1.1 10.6-2.4 10.6-7.7 0-3 2-1.7 6.3 3.8 6.8 9 6.9 14.7.2 19.1-7.5 4.9-13.6 2.5-21.6-8.4-2.2-2.9-4.4-5.3-4.9-5.3-1.4 0-1.2 6.5.4 13.2.8 3.6 1 5.8.3 5.8-1.4 0-12.1-5.5-18.1-9.4-6.9-4.4-16.3-14.1-17.6-18.2-2.2-6.6-23.3-17.2-36.1-18.2-6.1-.4-7.3-.1-18 4.3-29.3 12.3-46.1 16.2-65.8 15.2-6.4-.3-14.9-1.3-19-2.1-6.7-1.5-9.3-1.5-7.9-.1.9.9 23.3 5.5 34.3 7 11.5 1.7 20.3 1.8 26 .5 2.2-.5 4.1-.7 4.3-.6 1 1.1-1.5 6.1-5 10.1-7 8-25.4 15.3-47.3 18.6-8.7 1.4-10.3 1.5-30.2 2.5-25.4 1.1-19.5 3.7 8.7 3.8 18 .1 24-.3 24-1.5 0-.4-5.5-.6-12.2-.4l-12.3.4 16.1-2.4c8.9-1.3 19.8-3.3 24.3-4.4 10-2.5 22-8 27.7-12.6 4.4-3.6 8.4-10.5 8.4-14.4 0-2-.4-2.2-2.2-1.7-4 1.2-20.1 1.7-24.8.8l-4.5-.9 3.5-.6c1.9-.3 8.1-1.3 13.8-2.1 9.7-1.5 22.4-5.7 42-14 12.3-5.3 22.2-4.4 37.1 3.2 10 5.1 11.8 6.5 13.6 10.9 3.3 7.7 16 18.1 31.6 25.9l7.7 3.7 2.1 7.3c2.3 7.8 8.3 16.8 15.2 22.7 3.5 3.1 3.9 3.9 3.9 8.1 0 3.9-.5 5.3-2.9 7.6-3.9 3.9-7.3 4.6-13.5 2.7-8.7-2.8-18.1-14-24.6-29.4-1.2-2.9-3.1-5.5-4.1-5.8-3.3-1.1-14.9-13.7-21-22.9-4.9-7.4-6.8-9.4-11.9-12.4-5.8-3.4-10.1-7.5-15.9-15.2-3-4-7.1-3.9-8.5.3-1 2.6-.5 3.2 7.9 11.5 5 4.8 13.3 13.2 18.5 18.7 5.2 5.5 14.3 14.4 20.2 19.8 15.8 14.4 29.8 33 29.8 39.5 0 2.6-8.3 8.2-12.2 8.2-7.5 0-20-6.5-32-16.5-4-3.3-8.9-6.7-10.8-7.4-5.3-2.1-17.2-11.3-22.5-17.5-3.1-3.6-6.8-6.5-10.9-8.5-3.3-1.6-9-5.6-12.5-8.9-6.4-5.7-6.5-5.8-10.2-4.8-2.4.6-5.1.6-7.6 0-3.3-.8-4.9-.4-12 2.4-8.6 3.4-23.6 7.1-29 7.2-1.7 0-3.5.5-3.9 1.2-1.8 2.9 25.9-2.9 36.3-7.7 4.4-2 5.7-2.2 8.7-1.3 2.4.8 4.6.8 6.8.1 3.1-.9 3.7-.6 10.3 5 3.9 3.3 9.5 7.1 12.5 8.5 3 1.4 6.4 3.7 7.5 5.1 5.9 7.8 16 16 24 19.6 2.7 1.3 7.7 4.6 11 7.4 20.1 17.3 37.2 21.9 45.9 12.2 1.4-1.6 1.8-3.1 1.4-6.5l-.5-4.5 4.1 2.3c5.9 3.4 11.3 4.1 15.6 1.9 4.7-2.4 8-7.6 8-12.6 0-3.6-.7-4.7-7.6-11.9-7.4-7.8-11.5-14.7-12.7-21.4l-.6-3.1 7.3.5c8.7.5 12.9-.6 18.7-5.3 5.4-4.2 6.8-6.9 6.8-12.5.1-5.7-2.6-11.6-8.8-19.4-4.2-5.4-5.1-7.4-6.5-14.4-.9-4.5-1.9-8.9-2.2-9.7-.5-1.2.3-1.5 3.7-1.5 2.4 0 6.7-.8 9.4-1.9 8.5-3.1 11.3-3.3 13.6-1 1.5 1.5 1.9 2.8 1.5 4.7-.3 1.5-.9 5.7-1.2 9.4-.8 8.1.6 11 6.9 13.9 5.8 2.6 10.5 2.4 15-.6 7.4-5.1 8.5-11.5 3.5-21.3-3.8-7.4-3.3-9.1 3.3-10.9 4.5-1.2 8.8-1 17.2.8l4.7 1.1v10.1c0 11.3 1.2 16.9 4.1 19.8 2.5 2.5 5.3 2.4 9-.4 4.5-3.2 13.7-3.2 17.3.2 9 8.4 2.5 25.6-9 23.8-1.8-.3-4.6-1.5-6.2-2.7-2.4-1.8-3.2-1.9-5.4-1-1.4.7-3.5 2.9-4.7 5.1-2.5 4.4-4 13.9-2.2 13.9.6 0 1.1-1.6 1.1-3.8 0-4.7 3.2-11.7 6-13.2 1.8-.9 2.5-.8 4.1.6 2.4 2.1 11 3 14.7 1.5 4.2-1.8 7.2-7.7 7.2-14.2 0-12.8-10.5-18.4-22.7-12.3-2.8 1.5-5.9 2.4-6.8 2-3.1-1.2-4.5-7-4.5-18.9v-11.4l-5.8-1.7c-8.2-2.3-17.3-2.1-22.2.4-4.9 2.5-5 4.1-1 11.7 4.9 9.2 3.5 15.9-4.1 20.3-3.4 1.8-4.4 2-7.9 1-2.2-.6-5.5-2.3-7.2-3.9-3.4-3-3.6-4-2.3-16.5.7-7.1.6-8-1-9.2-2.7-2-8.6-1.7-14.7.6-5.6 2.1-14.4 2.8-13.3 1 .3-.5 2.3-1.3 4.4-1.6l3.8-.6.7-7.2c.3-3.9 1.5-10.9 2.7-15.6 3.2-13.5 3.6-20.8 1.1-25.6-5.1-10-25.7-24.6-54.7-38.9-22.8-11.3-36.5-16.8-46.8-18.9-9.5-2-13-1.9-23.7.4zm33.8 57.7c1.8 1.4 3.1 2.8 2.8 3.1-.9.8 5.9 6.7 10.2 8.8 5.4 2.6 11.9 2.5 14.2-.3l1.8-2.2 1.1 4.4c1.3 5.1 4.1 8.5 7.1 8.5 2.5 0 7.3-2.4 11.7-5.8l3.3-2.4-.8 3.3c-.4 1.9-.7 7-.7 11.4v8l-13.9-7.2c-7.7-3.9-17.8-9.8-22.5-13.2-9.8-6.9-25.1-19.8-25.1-21.1 0-1.3 6.9 1.7 10.8 4.7zm71.1 52.4c5 1 5.1 1.2 6.6 6.2.9 2.9 1.9 7.4 2.2 10.2.5 4 1.4 5.9 4.9 9.8 8.7 9.8 11.9 20.8 7.7 26.5-3.1 4.2-9.4 8.1-14.5 9.1-4.9.9-13.5-.2-15.1-1.8-.9-.9-4.9-16.2-4.4-16.8.2-.1 2.6 2.5 5.3 6 5.7 7.1 10.4 9.5 16.9 8.3 5.5-1.1 10.5-6.3 10.5-11 0-4.5-3.6-11.7-8.7-17.6-2.7-3-4.4-6.2-5.2-9.7-1-4-2.8-7.1-7.2-12.3-3.2-3.9-5.9-7.3-5.9-7.5 0-.7 1.4-.6 6.9.6zm5.6 30.3c-.8 3.7-3.9 6.6-7.1 6.6-3.4 0-3.6-1.3-1-4.9 1.3-1.8 3-4.6 3.7-6.3l1.3-3.1 1.8 2.3c1.4 1.7 1.7 3.2 1.3 5.4zm-88.7-3.2c2.3 4.5 9.6 11.4 15.8 14.9 4 2.3 6.3 4.7 10.5 11.1 6.1 9.3 15.4 19.7 20.4 22.8 2.7 1.6 6.5 7.8 6.5 10.5 0 .7-19.1-17.9-40.1-39-10.9-11-19.9-20.5-19.9-21.1 0-3.7 4.7-3.1 6.8.8zM519.1 96.1c-12 1.4-20.9 6.4-39.6 22-5.5 4.6-13.1 10.6-17 13.4-3.8 2.7-9.2 7.1-11.8 9.7-2.6 2.6-5.3 4.8-5.8 4.8-2.1 0-6.9-3.4-6.9-4.9 0-.8 1.2-2.9 2.6-4.8 1.4-1.9 2.3-3.7 1.9-4.1-.4-.4-1-.1-1.2.6-.3.7-1.1 1.9-1.9 2.7-.8.7-1.7 2.7-2 4.4-.6 2.9-.1 3.8 4.9 7.4 1.5 1 1.6 1.7.8 4-1.6 4.1-1.3 9.9.4 12.4.9 1.3 3.5 2.8 5.8 3.4 2.3.6 4.8 1.3 5.6 1.5.9.3-.1 1.3-2.5 2.6-4.3 2.3-7.6 4.7-13.1 9.3-4.5 4-4.7 8.3-.6 15.7 5.5 9.8 15.3 15.2 29.6 16.4l7.8.7-.6 4.9c-1.3 9.2-1.8 10.3-5.4 10.6-1.8.2-4 .1-4.8-.3-1.4-.5-1.2-.9.7-3 1.3-1.4 2.1-2.7 1.9-3-2.8-2.7-45.1-3-69.8-.5-7.6.8-10.7.8-11.5 0-2-2-1.1-12.4 1.3-16.4 2.9-4.7 5.7-5.3 10.5-2.5 4.6 2.7 11.4 2.9 15 .6 6.8-4.5 10.3-17.8 6.2-23.9-3.2-4.8-14.8-7.7-17.6-4.3-2.1 2.5-10.3 2-13.7-.9-2.6-2.3-2.8-2.9-2.8-10.3 0-4.3.7-9.5 1.4-11.6l1.3-3.8 6.3.6c12.3 1.4 26.5-2.9 27.3-8.1.2-1.1-1.1-4.6-2.8-7.8-1.6-3.2-3-6.5-3-7.3 0-2.7 3.1-7.3 6-8.8 3.4-1.8 10.9-2 14.3-.4 2.9 1.3 5.7 5.8 5.7 9.3 0 1.4.5 2.6 1 2.6 1.5 0 1.2-4.7-.5-8.1-4.5-8.7-19-9.4-25.4-1.3-2.9 3.7-2.7 9.2.6 15.8 1.4 2.9 2.3 6.1 2 6.9-.3.9-3.7 2.5-7.4 3.7-5.4 1.7-8.7 2.1-16.2 1.8l-9.5-.3-1.4 4.5c-1.8 5.7-2 18.8-.4 21.7 2.9 5.3 14.1 7.3 17.8 3.2 2.1-2.3 6.3-2 11.7.8 7.1 3.6 7 14.9-.1 22.5-3.9 4.2-8.3 4.6-14.8 1.3-4.8-2.4-5.7-2.6-8.4-1.5-4.2 1.7-7 8.1-7 15.6 0 9 .7 9.6 11 8.4 16.6-1.9 46.4-2.9 57.8-2.1 6.2.5 11.5 1 11.7 1.3.2.2-.4 1.3-1.3 2.5-1.3 1.6-3.5 2.5-8.6 3.3-12.9 2-16.4 5.6-12.5 13.1 1.2 2.4 2.4 3.9 2.6 3.2.2-.6-.3-2.5-1.1-4.2-.9-1.6-1.6-3.7-1.6-4.6 0-1.8 6.4-4.9 12.4-6 3.8-.6 3.8-.6 3.2 1.9-.8 3.3.1 3.3 2.6 0 1.9-2.4 2.4-2.5 8.3-1.9l6.3.7 1.1-5.1c.6-2.8 1.4-6.8 1.7-9 .6-3.7.8-3.9 5.3-4.4 12.9-1.6 20.9-1.8 25.1-.7 4.8 1.2 12 5.4 12 7 0 .5-3.9 2.8-8.7 5.2-6.3 3-10.6 6-15.2 10.5-4 4-8.8 7.5-13 9.5-7.8 3.8-7.6 3.6-6.8 4.4.3.3 3-.5 6-1.9l5.4-2.5-.5 4.4c-.4 3.6 0 5 2.2 7.9 2.5 3.2 3.1 3.5 8.1 3.5 4.5 0 6.9-.8 14.2-4.6 4.8-2.5 12.5-5.6 17.1-7 4.6-1.3 10.2-3.4 12.5-4.6 3.2-1.7 5.6-2.1 9.8-1.9 3.1.1 6-.2 6.3-.8.4-.5-.8-2.7-2.6-4.7l-3.3-3.8 3.5 2c1.9 1 8.3 6.7 14.2 12.5 11.1 10.9 14.6 12.5 5.9 2.8-10.6-11.9-36.1-29.6-36.1-25 0 .5.9 1.2 1.9 1.6 2.6.8 14.2 12.8 13.3 13.7-.4.4-2.5.3-4.6-.2-3.4-.7-4.6-.5-9.1 2.1-2.8 1.6-8.6 3.8-12.9 4.9-4.3 1.1-10.3 3.5-13.4 5.4-8 4.8-16.5 7.1-20.2 5.6-4.1-1.7-6.6-7.2-5-11.1 3.1-7.3 23.7-20.1 36.9-23 15.1-3.2 27.4-14.1 35.5-31.3 4.4-9.2 5.3-13.6 6.5-29.2 1.1-13.5 3.4-23.1 7.8-31.7 1.3-2.7 2.2-5.4 2-6-.7-2-6.7 11.1-8.7 18.9-1.1 4.1-2.4 13-3 19.8-.6 6.8-1.7 15-2.6 18.3-3.1 11.9-14.4 27.9-24.1 34.1-2.6 1.7-9 4.2-14.2 5.7-5.1 1.4-12.4 4.2-16.2 6.1-3.8 1.9-6.9 3-6.9 2.5 0-.4 1.9-1.6 4.3-2.5 5.9-2.2 16.7-8.7 16.7-10 0-1.2-4.4-4.2-10.5-7.2-4.5-2.1-13.8-2.5-26.4-1l-7.4.8.7-5.8c.4-3.3 1-13.8 1.2-23.4l.4-17.5 7.1-.3c5.4-.2 10.4.5 22.1 3.4 14 3.4 15.8 3.6 27.1 3.2 10.7-.3 12.8-.7 16.6-2.8 2.5-1.4 4.6-3 4.8-3.6.2-.6-1.7-4.2-4.3-7.9-2.6-3.8-5.3-8.9-6.1-11.4-1.2-4-1.2-5.2.2-9.8 1.6-5 1.4-7.9-.5-6.8-.4.3-1.1 2.3-1.5 4.4-.4 2.2-1.7 4.9-3.2 6.2-2.2 2.2-3.4 2.4-11.7 2.4-10.4 0-14.2-1.6-15.1-6.3-1.7-8.2-2.3-8.8-2.4-2.5-.1 3.4-.5 7.2-1 8.5-1.8 4.5-7 8.5-15.4 11.7-9.9 3.7-10.3 3.7-11.1.8-.6-2.1-.7-2.1-3.2 1-1.4 1.8-2.9 3.3-3.3 3.3-.4 0-1.5-2.6-2.4-5.8-2.5-8.8-2.2-8.6-11.7-6.8-4.5.9-10.5 1.6-13.3 1.6h-5.1l4.4-4.5c2.5-2.5 7.9-6.9 12-9.7 4.1-2.9 10.4-7.8 14-11 16.7-14.7 29-22.1 39.5-23.6 4.5-.7 90.8-.9 93.5-.2.3 0-1.9 1.1-4.9 2.3-3 1.3-5.6 2.8-5.9 3.5-.2.7 1.8.2 5-1.2 2.9-1.3 6.8-2.7 8.6-3.1 7.2-1.5 1.8-4-8.8-3.9-3.6 0-18.2.1-32.5.3-14.3.2-30.5.1-36-.2-5.5-.3-13.3-.1-17.4.4zm0 45.3c7.2 3 18 3 25 .2 1.3-.6 1.8.1 2.3 3.3.3 2.3 2.6 6.9 5.6 11.2 4.5 6.8 4.8 7.5 3.3 9.1-4.5 5-29.2 5-48.7-.1-8.1-2.1-14.4-3.1-19.3-3.1-4 0-7.3-.4-7.3-.8 0-2.2 4.8-4.9 7.8-4.4 4.1.6 15-3.2 20.2-6.9 2.2-1.6 4.8-4.5 5.7-6.4.9-1.9 1.8-3.5 1.9-3.5.1 0 1.7.6 3.5 1.4zm-43.8 9.3c3.2 10 3.2 9.7-2.6 12.9-4.6 2.6-6.2 2.9-13.9 2.9-7.2 0-9.1-.3-11.5-2.1-3.2-2.4-4.2-7.8-2.3-12.4 1.1-2.6 1.5-2.7 10.3-3.4 5.1-.4 10.8-1.1 12.7-1.5 5.6-1.4 5.8-1.3 7.3 3.6zm3.3 33.5c-.3 10.6-.9 21-1.2 23.2l-.7 3.9-7.6-.6c-4.2-.3-9.9-1.3-12.8-2.1-10.6-3.3-21.1-16.9-18.3-23.7.7-1.5 1.9-3.4 2.6-4 1.2-1 1.4-.3.8 4.5-.5 4.1-.2 6.7 1 10 2.5 6.6 4.5 7.8 12.9 8.4 9.4.6 10.8-.7 7.6-6.7-1.8-3.5-2.1-5.4-1.8-12.4l.4-8.2-5.7-.3c-9.5-.5-7.5-2.1 9.7-7.8 15.2-5.1 13.9-6.7 13.1 15.8zm-19.4 1.8c-.4 6.7-.1 8.5 1.7 11.5 1.2 2 2.1 3.8 2.1 4.1 0 1-11.7.3-14-.9-5.3-2.6-7.5-12.1-4.6-19.2l1.5-3.5h13.6l-.3 8z"
            />
            <path
              d="M613.2 105.1c-5.3.9-9.8 2.7-8.9 3.5.2.2 3.3-.2 6.8-1.1 12.5-2.9 11.5-3.6 21.9 14 5 8.6 13.5 24.7 18.9 35.9 10.5 21.6 15.3 29.5 22.1 36.3 6.1 6.2 7.1 5.3 1.3-1.1-7.6-8.3-12-15.5-21.3-35.1-4.7-9.9-13.2-26-18.9-35.8-11.6-19.9-9.9-18.6-21.9-16.6zM592.9 107.9c-1.3 1.1-1.9 2-1.3 2.1 1.5 0 5-2.7 4.3-3.4-.3-.4-1.7.2-3 1.3zM599 110c-.8.5-1.2 1.2-.9 1.5.8.8 3.9-.5 3.9-1.6s-1.1-1.1-3 .1zM590.6 116.6c-1.4 1.5-2.6 3.1-2.6 3.7 0 .6 1.6-.4 3.5-2.3s3.5-3.5 3.5-3.7c0-1-2.1.1-4.4 2.3zM582.7 117.8c-2.1 2.4-2.4 5-.3 3.3 1.2-1 2.9-5.1 2.2-5.1-.2 0-1 .8-1.9 1.8zM584.6 124.6c-.9 2.4-.7 2.8.9 1.4 1.7-1.4 2-3 .6-3-.5 0-1.2.7-1.5 1.6zM581.8 129.5c-4.3 7-6 33.7-3.5 52.5 3.7 26.7 17.2 64.4 30.1 83.9 6.7 10 23 25.1 49.1 45.4 5.5 4.3 10.1 8.2 10.3 8.7.2.4-1.4.4-3.5 0-2.1-.5-7.2-.6-11.3-.3-8.8.6-15.9-1.4-26.2-7.4-12-7-20.9-17.4-38.3-45-2.2-3.5-4.6-6.3-5.2-6.3-.7 0-.1 1.8 1.6 4.2 1.6 2.4 6.7 10.1 11.4 17.3 13 19.7 25.2 30.9 39.7 36.4 5.4 2 8.4 2.4 18.5 2.5 10.9.1 12.6.4 18.3 3 3.4 1.6 6.5 2.7 6.7 2.4.3-.3-3.9-3.8-9.2-7.9-26.3-19.9-42.2-33.5-53.5-45.7-6.5-7-8.7-10.4-14.6-22.5-11.8-23.9-18.9-46.7-22.8-73.2-2-13.5-1.4-28.8 1.6-39.4 1.9-6.7 2.3-11 .8-8.6zM307.4 214.3c-8.2 3.1-12.4 9.6-10.2 15.8.6 1.6.8 3.2.5 3.5-.3.3-5.8 0-12.2-.6-6.5-.6-11.9-.9-12-.8-.1.2-.5 3.8-.7 8l-.5 7.7 3.1 1.1c1.7.7 6.7 1.4 11.2 1.7 9.4.6 14.7-.7 17.1-4.3 1.5-2.3 1.4-3.7-.2-7.7-.6-1.5.1-1.7 7.4-1.7h8.2l1.5 3.5c.8 1.9 2.4 4.1 3.7 4.9 7.2 4.6 24.5 7 31 4.3 4.3-1.8 4.6-3.6 1.6-7.9-1.2-1.6-2-3-1.8-3.2.2-.2 5.7-.8 12.2-1.4 8.9-.9 14.4-2 22.5-4.8 6.2-2.1 13.5-3.9 17.3-4.2 6.4-.4 6.6-.4 11.1 3.6l4.6 3.9-2.4 3.9c-3.7 6.1-2.6 14.9 2.8 21.5 4.8 5.8 16.4 4.7 21.5-2 2.9-3.8 1-4-2.7-.2-3.6 3.7-4.5 4.1-8.9 4.1-6.3 0-9.4-2.4-12.1-9.6l-2.1-5.6 2.6-5.5c1.4-3.1 2.5-6.1 2.5-6.8 0-2.2-7.8-8.4-11.4-9.1-4.8-.9-11.5.2-20 3.3-10 3.7-17.5 5.2-29.7 5.9l-10.6.7-1.1-3.2c-1.4-3.9-2.6-4.1-1.9-.2.7 3.5-.6 4.2-7.1 3.7-2.3-.2-4.5.1-4.8.5-.3.5 2.7.9 6.6.9 6.7 0 7.2.2 10 3.2 1.6 1.8 3 3.9 3 4.6.1 6-30.6 2.5-34-3.9-1.9-3.4-1.1-3.9 6-3.9 4 0 7-.4 7-1s-2.6-1-5.9-1h-5.8l1.9-3.9c1.8-3.7 1.8-4.1.3-7.7-2.1-5.1-4.2-7.3-9-9.5-4.6-2.1-5.8-2.2-10.1-.6zm9.4 2.1c4.3 2.3 8.2 7 8.8 10.7.5 2.8.1 3.9-2.3 6.1-2.8 2.6-3 2.6-12.3 2l-9.5-.7-1.9-3.9c-1.9-3.9-1.9-4 0-7.5 2-3.8 8.5-8.1 12.2-8.1 1.3 0 3.5.6 5 1.4zm-24.6 19.2c6.6.6 8 1 9.3 3 4.8 7.4-3.6 12.4-17.4 10.3-9.5-1.5-9.7-1.6-9.3-8.9l.4-6.2 4.6.6c2.6.3 8.2.8 12.4 1.2zM456 239.9c0 1.1.5 2.3 1 2.6.6.4 1-.5 1-1.9s-.4-2.6-1-2.6c-.5 0-1 .9-1 1.9zM460 245.5c0 .2 1.4 1.1 3 2 1.7.9 3 1.2 3 .7s-1.1-1.4-2.5-2c-2.8-1.3-3.5-1.4-3.5-.7zM469 248.4c0 1 2.8 1.3 4.1.5 1-.6.7-.9-1.3-.9-1.6 0-2.8.2-2.8.4zM581 258.5c0 .8.4 1.5 1 1.5.5 0 .7-.7.4-1.5-.4-.8-.8-1.5-1-1.5-.2 0-.4.7-.4 1.5z"
            />
          </svg>
        </div>
        <button
          className="bg-orange rounded-md text-xs w-44 h-10 scale-125 text-[#FFFFFF] inline-block bottom-[20px] xl:left-[40%] lg:left-[20%] hover:scale-[1.35] ease-in-out transition-all"
          onClick={[Function]}
          type="button"
        >
          Ask Question +
        </button>
      </div>
    </section>
    <button
      className=" lg:hidden mt-10 text-xs ml-5 sm:text-3xl mx-auto bg-orange rounded-md sm:w-[14rem] sm:h-[4rem] w-[8rem] h-[2rem] text-[#FFFFFF] absolute inline-block  left-[30%]"
      onClick={[Function]}
      type="button"
    >
      Ask Question +
    </button>
    <div
      className="mt-32 md:mr-7 lg:mr-10 pb-10"
    />
  </div>
</div>
`;
